### PR TITLE
 updated assignment with answers

### PR DESCRIPTION
**Pull request by priyankapatki**

### CODE REVIEW

File: D04_ex00.py [hunk 1]
```diff
@@ -11,19 +11,38 @@
...
+        try:
+            user_input = int(input('Please guess a random number between 1 and 25 = '))
+            if (user_input == num):
+                print('You got it right!')
+                count += 1
+                break
+            elif(user_input < num):
+                print('The number was too low.')
+                count += 1
+                continue
```
Comment (danielsgriffin) [inline D04_ex00.py]: are these continues necessary?

File: HW04_ch07_ex02.py [hunk 2]
```diff
@@ -24,12 +24,27 @@
...
+
+        except:
+            print(" Not a valid expression ")
 
 ###############################################################################
 def main():
-    pass  # Remove this line and uncomment below once eval_loop is defined.
-    # eval_loop()
+    
+    a = eval_loop()
```
Comment (danielsgriffin) [inline HW04_ch07_ex02.py]: What if 'done' was the first thing input?

File: HW04_ch08_ex00.py [hunk 3]
```diff
@@ -18,16 +18,32 @@
...
 # Imports
 
-
 # Body
 
+def count(word, letter_arg):
+    letter_count = 0
+    for letter in word:
+        if letter == letter_arg:
+            letter_count = letter_count + 1
```
Comment (danielsgriffin) [inline HW04_ch08_ex00.py]: letter_count += 1

File: HW04_ch08_ex05.py [hunk 4]
```diff
@@ -1,4 +1,63 @@
...
+            nc = z - (a - nc) + 1
+        elif nc > z:
+            # rotate  right          
+            nc = a + (nc - z) - 1
+        new_word += chr(nc)
+    
+    return new_word
+
+
+def main():
```
Comment (danielsgriffin) [inline HW04_ch08_ex05.py]: Good tests here.

**Comment by danielsgriffin** [comment]
Good job.

File: HW04_ch07_ex02.py [hunk 5]
```diff
@@ -24,12 +24,27 @@
...
+    answer = "'none'"      # initializing variable 'answer'
+    while user_input != 'done':   # loop will continue till the user enters done  as the input
+        user_input = input('Please enter the expression to be evaluated: ')
+        try:
+            if user_input != 'done':
+                answer = eval(str(user_input))
+                print('The answer is :' + str(answer))
+            else:
+                return answer
+                break
```
Comment (jollylst) [inline HW04_ch07_ex02.py]: is "break" here after return necessary?

File: D04_ex00.py [hunk 6]
```diff
@@ -11,19 +11,38 @@
...
+            else:
+                print('The number was too high.')
+                count += 1
+                continue
+        except:
+            print('Not a number.')
+            count += 1
```
Comment (Nic-Mon) [inline D04_ex00.py]: You can have only one `count += `1` if you put it outside of the try block

File: HW04_ch07_ex02.py [hunk 7]
```diff
@@ -24,12 +24,27 @@
 
 
 # Body
-
+def eval_loop():
+    user_input = ''
+    answer = "'none'"      # initializing variable 'answer'
```
Comment (Nic-Mon) [inline HW04_ch07_ex02.py]: Do you need two sets of quotes?

File: HW04_ch08_ex05.py [hunk 8]
```diff
@@ -1,4 +1,63 @@
...
 # Structure this script entirely on your own.
 # See Chapter 8: Strings Exercise 5 for the task.
 # Please do provide function calls that test/demonstrate your
 # function.
+
+###############################################################################
+
+
+# Find the ord('a') and ord('z') based on the case of the character c
+def limit(c):
```
Comment (Nic-Mon) [inline HW04_ch08_ex05.py]: nice use of smaller functions for clarity

**Pull request closed without merging**